department-of-veterans-affairs/vets-api
# Api 34592 poa v2 claimant signatures

## Summary

- This work is behind a feature toggle (flipper): NO
- Removes claimant firstName, middleInitial, lastName from claimant section of request, and adds claimantId.
- Adds logic to detect cases where a claimantId (ICN) is supplied and lookup the ICN in MPI, then use that data to sign the pdf with claimant's firstName and lastName.
- Adds logic to validate that a claimantId is provided when there is other data in the claimant section of the request. Note that in the case where claimantId is provided without other required fields in the claimant section of the request, existing validation raises error(s) indicating the missing claimant attributes.


## Related issue(s)

[API-34592](https://jira.devops.va.gov/browse/API-34592)

## Testing done

- Submit 2122 & 2122a requests with claimantId & claimant information supplied and confirm that the claimant's firstName & lastName is used to sign the PDF in box 22A.
- Submit 2122 & 2122a requests without claimantId & claimant information supplied and confirm that the veteran's firstName & lastName is used to sign the PDF in box 22A.
- Submit 2122 & 2122a requests without claimantId but with other claimant information supplied and confirm that a 422 is raised with error message indicating the claimantId is required.
- Submit 2122 & 2122a requests with claimantId but without other claimant information supplied and confirm that a 422 is raised with error message(s) indicating the missing field(s) is/are required.


## What areas of the site does it impact?
modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb
modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb
modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json
modules/claims_api/config/schemas/v2/2122.json
modules/claims_api/config/schemas/v2/2122a.json
modules/claims_api/spec/requests/v2/veterans/power_of_attorney_ind_request_spec.rb
modules/claims_api/spec/requests/v2/veterans/power_of_attorney_org_request_spec.rb
modules/claims_api/spec/sidekiq/v2/poa_form_builder_job_spec.rb

## Acceptance criteria
